owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/filelist/MainFileListFragment.kt

```diff
@@ -196,7 +227,142 @@ class MainFileListFragment : Fragment(), SortDialogListener, SortOptionsView.Sor
+        } else {
+            binding.fabMain.visibility = View.GONE
+        }
+    }
+
+    /**
+     * Adds labels to all mini FABs.
+     */
+    private fun setFabLabels() {
+        binding.fabUpload.title = resources.getString(R.string.actionbar_upload)
+        binding.fabMkdir.title = resources.getString(R.string.actionbar_mkdir)
+    }
+
+    /**
+     * Removes the labels on all known min FABs.
+     */
+    private fun removeFabLabels() {
+        binding.fabUpload.title = null
+        binding.fabMkdir.title = null
+        ((binding.fabUpload.getTag(com.getbase.floatingactionbutton.R.id.fab_label)) as TextView).isVisible = false
+        ((binding.fabMkdir.getTag(com.getbase.floatingactionbutton.R.id.fab_label)) as TextView).isVisible = false
+    }
+
+    /**
+     * registers all listeners on all mini FABs.
+     */
+    private fun registerFabListeners() {
+        // TODO Register upload listener
+        registerFabMkDirListeners()
+    }
+
+    /**
+     * Registers [android.view.View.OnClickListener] and [android.view.View.OnLongClickListener]
+     * on the 'Create Dir' mini FAB for the linked action and [Snackbar] showing the underlying action.
+     */
+    private fun registerFabMkDirListeners() {
+        binding.fabMkdir.setOnClickListener {
+            val dialog = CreateFolderDialogFragment.newInstance(mainFileListViewModel.getFile(), this)
+            dialog.show(requireActivity().supportFragmentManager, DIALOG_CREATE_FOLDER)
+            binding.fabMain.collapse()
+            recordMiniFabClick()
+        }
+        binding.fabMkdir.setOnLongClickListener {
+            showSnackMessage(R.string.actionbar_mkdir)
+            true
+        }
+    }
+
+    /**
+     * records a click on a mini FAB and thus:
+     *
+     *  1. persists the click fact
+     *  1. removes the mini FAB labels
+     *
+     */
+    private fun recordMiniFabClick() {
+        // only record if it hasn't been done already at some other time
+        if (!miniFabClicked) {
+            val sp = android.preference.PreferenceManager.getDefaultSharedPreferences(activity)
+            sp.edit().putLong(KEY_FAB_EVER_CLICKED, 1).apply()
+            miniFabClicked = true
+        }
+    }
+
+    /**
+     * Show a temporary message in a Snackbar bound to the content view of the parent Activity
+     *
+     * @param messageResource Message to show.
+     */
+    private fun showSnackMessage(messageResource: Int) {
+        val snackbar = Snackbar.make(
+            requireActivity().findViewById(R.id.coordinator_layout),
+            messageResource,
+            Snackbar.LENGTH_LONG
+        )
+        snackbar.show()
+    }
+
+    private fun showSnackMessage(messageResource: String) {
+        val snackbar = Snackbar.make(
+            requireActivity().findViewById(R.id.coordinator_layout),
+            messageResource,
+            Snackbar.LENGTH_LONG
+        )
+        snackbar.show()
```

Comment:
I see we have two methods to display a SnackMessage, could we make an extension function (Thanks to kotlin :D) where we just pass the message? We could reuse it in any fragment.

Reply:
We have an existing extension. Used!